### PR TITLE
Update and simplify vimrc

### PR DESCRIPTION
**Pull request by anassinator**
This greatly simplifies our default `vimrc` by removing most plugins and documenting what each setting does. This also moves to using `vim-plug` instead of the outdated `Vundle.vim`. Documentation of our `vim` configuration was also added to the `README.md`

### CODE REVIEW

File: update [hunk 1]
```diff
@@ -9,7 +9,7 @@ section=$(tput bold; tput setaf 2)
 reset=$(tput sgr0)
 
 echo "${section}Updating apt dependencies...${reset}"
-sudo apt-get -qq update
+sudo apt-get -q update
```
Comment (foochuanwei) [inline update]: Why?

File: setup/config/vimrc [hunk 2]
```diff
@@ -1,140 +1,159 @@
...
+  Plug 'taketwo/vim-ros'
+  Plug 'plasticboy/vim-markdown'
+  Plug 'sudar/vim-arduino-syntax'
+  Plug 'vim-syntastic/syntastic'
 
-" Quick access to vimrc
-nmap <silent> ,ev :e $MYVIMRC<cr>
-nmap <silent> ,sv :so $MYVIMRC<cr>
+  " Autocomplete
+  Plug 'lifepillar/vim-mucomplete'
```
Comment (foochuanwei) [inline setup/config/vimrc]: Are we missing `let g:mucomplete#enable_auto_at_startup = 1`, or is the omission intentional?

**Comment by foochuanwei** [comment]
Other useful settings you might find helpful:

```vim
" Raise a dialog instead of failing for some operations. e.g. `:q` or `:e` when there are unsaved buffers.
set confirm

" Show tabs as characters.
set list
set listchars=tab:!-  " Show tabs as !---.
```

File: update (hunk 1 above)
Comment (anassinator) [reply update]: Yeah to increase verbosity because it kept looking like it was hanging. Fixed.

File: setup/config/vimrc (hunk 2 above)
Comment (anassinator) [reply setup/config/vimrc]: It's intentional, that setting autocompletes without pressing `<TAB>`, which is counter-intuitive for most people I believe. Omitting it means suggestions and autocompletions only occur on a `<TAB>` press, which is more in-line with other editors/IDEs.

**Comment by anassinator** [comment]
I added your suggestions as well, thanks!

**Pull request merged**